nautobot/core/settings.py

```diff
@@ -914,3 +922,35 @@ def UI_RACK_VIEW_TRUNCATE_FUNCTION(device_display_name):
     "TimeZoneSerializerField": {"type": "string"},
     "UUIDField": {"type": "string", "format": "uuid"},
 }
+
+
+#
+# django-silk is used for optional request profiling for debugging purposes
+#
+
+SILKY_PYTHON_PROFILER = True
+SILKY_PYTHON_PROFILER_BINARY = True
+SILKY_PYTHON_PROFILER_EXTENDED_FILE_NAME = True
+SILKY_ANALYZE_QUERIES = False  # See the docs for the implications of turning this on https://github.com/jazzband/django-silk?tab=readme-ov-file#enable-query-analysis
+SILKY_AUTHENTICATION = True  # User must login
+SILKY_AUTHORISATION = True  # User must have permissions
+
+
+# This makes it so that only superusers can access the silk UI
+def silk_user_permissions(user):
+    return user.is_superuser
+
+
+SILKY_PERMISSIONS = silk_user_permissions
```

Comment:
Nitpick incoming

```suggestion
SILKY_PERMISSIONS = lambda user: user.is_superuser
```

Reply:
Ha, if you can tell me the right way to suppress pylint and ruff here, I am fine with this. I will find several commits from me trying to do it...

Reply:
PEP8 makes a good case for declaring the function explicitly so we should probably leave it as-is